redash/handlers/data_sources.py

```diff
@@ -25,7 +25,14 @@ class DataSourceResource(BaseResource):
     @require_admin
     def get(self, data_source_id):
         data_source = models.DataSource.get_by_id_and_org(data_source_id, self.current_org)
-        return data_source.to_dict(all=True)
+        ds = data_source.to_dict(all=True)
+        page = 'admin/data_source/' + str(ds.id)
+        self.record_event({
+            'action': 'view',
+            'object_id': page,
+            'object_type': 'api_call',
```

Comment:
Here the `object_type` should be `datasource` and `object_id` should be `ds.id`.